exp/controllers/awsmachinepool_controller.go

```diff
@@ -404,9 +426,16 @@ func (r *AWSMachinePoolReconciler) updatePool(machinePoolScope *scope.MachinePoo
 		"subnets of machinePoolScope", subnetIDs,
 		"subnets of existing asg", existingASG.Subnets)
 	less := func(a, b string) bool { return a < b }
-	subnetChanges := cmp.Diff(subnetIDs, existingASG.Subnets, cmpopts.SortSlices(less)) != ""
+	subnetChanges := cmp.Diff(subnetIDs, existingASG.Subnets, cmpopts.SortSlices(less))
+	if subnetChanges != "" {
+		r.Recorder.Eventf(machinePoolScope.AWSMachinePool, corev1.EventTypeNormal, "ASGDiffDetected", subnetChanges)
+	}
 
-	if asgNeedsUpdates(machinePoolScope, existingASG) || subnetChanges {
+	asgDiff := detectASGDrift(machinePoolScope, existingASG)
+	if asgDiff != "" {
+		r.Recorder.Eventf(machinePoolScope.AWSMachinePool, corev1.EventTypeNormal, "ASGDiffDetected", asgDiff)
```

Comment:
Having thought more about it, I'm not sure an Event is what we want. Events have a 1 hour TTL by default.

Is our goal to record this information _always_, or only if the user is asking for more verbose logs?

If it's the latter, then we could use `machinePoolScope.Debug`, as we do in other places.

Reply:
Also, Kubernetes drops events if the queue is full. So events should be something that is okay to be missed.

Reply:
Okay, I don't have strong opinions here, so I swapped back to `machinePoolScope.Debug`